examples/nlp/text_extraction_with_bert.py

```diff
@@ -0,0 +1,323 @@
+        for para in item["paragraphs"]:
+            context = para["context"]
+            for qa in para["qas"]:
+                question = qa["question"]
+                answer_text = qa["answers"][0]["text"]
+                all_answers = [_["text"] for _ in qa["answers"]]
+                start_char_idx = qa["answers"][0]["answer_start"]
+                squad_eg = SquadExample(
+                    question, context, start_char_idx, answer_text, all_answers
+                )
+                squad_eg.preprocess()
+                squad_examples.append(squad_eg)
+    return squad_examples
+
+
+def create_inputs_targets(squad_examples):
+    dataset_dict = {
+        "input_ids": [],
+        "token_type_ids": [],
+        "attention_mask": [],
+        "start_token_idx": [],
+        "end_token_idx": [],
+    }
+    for item in squad_examples:
+        if item.skip == False:
+            for key in dataset_dict:
+                dataset_dict[key].append(getattr(item, key))
+    for key in dataset_dict:
+        dataset_dict[key] = np.array(dataset_dict[key])
+
+    x = [
+        dataset_dict["input_ids"],
+        dataset_dict["token_type_ids"],
+        dataset_dict["attention_mask"],
+    ]
+    y = [dataset_dict["start_token_idx"], dataset_dict["end_token_idx"]]
+    return x, y
+
+
+train_squad_examples = create_squad_examples(raw_train_data)
+x_train, y_train = create_inputs_targets(train_squad_examples)
+print(f"{len(train_squad_examples)} training points created.")
+
+eval_squad_examples = create_squad_examples(raw_eval_data)
+x_eval, y_eval = create_inputs_targets(eval_squad_examples)
+print(f"{len(eval_squad_examples)} evaluation points created.")
+
+"""
+Create the Question Answering Model using BERT and Functional API
+"""
+
+
+def create_model():
+    ## BERT encoder
+    encoder = TFBertModel.from_pretrained("bert-base-uncased")
+
+    ## QA Model
+    input_ids = layers.Input(shape=(max_len,), dtype=tf.int32)
+    token_type_ids = layers.Input(shape=(max_len,), dtype=tf.int32)
+    attention_mask = layers.Input(shape=(max_len,), dtype=tf.int32)
+    embedding = encoder(
+        input_ids, token_type_ids=token_type_ids, attention_mask=attention_mask
+    )[0]
+
+    start_logits = layers.Dense(1, name="start_logit", use_bias=False)(embedding)
+    start_logits = layers.Flatten()(start_logits)
+
+    end_logits = layers.Dense(1, name="end_logit", use_bias=False)(embedding)
+    end_logits = layers.Flatten()(end_logits)
+
+    start_probs = layers.Activation(keras.activations.softmax)(start_logits)
+    end_probs = layers.Activation(keras.activations.softmax)(end_logits)
+
+    model = keras.Model(
+        inputs=[input_ids, token_type_ids, attention_mask],
+        outputs=[start_probs, end_probs],
+    )
+    loss = keras.losses.SparseCategoricalCrossentropy(from_logits=False)
+    optimizer = keras.optimizers.Adam(lr=5e-5)
+    model.compile(optimizer=optimizer, loss=[loss, loss])
+    return model
+
+
+use_tpu = True
```

Comment:
Please add a paragraph of text explaining that this example should preferably be run on the Colab TPU runtime.